Dockerfile

```diff
@@ -0,0 +1,94 @@
+FROM quay.io/pypa/manylinux_2_28_x86_64
+
+# When run the container, please map the project folder to container folder
+
+RUN yum -y install rdma-core-devel libuv-devel
+
+RUN dnf install epel-release
+RUN curl -O https://developer.download.nvidia.com/compute/cuda/12.8.0/local_installers/cuda-repo-rhel8-12-8-local-12.8.0_570.86.10-1.x86_64.rpm
+RUN rpm -i cuda-repo-rhel8-12-8-local-12.8.0_570.86.10-1.x86_64.rpm
+RUN dnf clean all
+RUN dnf makecache
+
+# install cuda
+RUN dnf -y install cuda-toolkit-12-8
+
+# build spdlog
+WORKDIR /tmp
+RUN git clone --branch v1.15.1 --recurse-submodules https://github.com/gabime/spdlog.git
+WORKDIR /tmp/spdlog
+RUN mkdir build
+WORKDIR /tmp/spdlog/build
+RUN cmake ..
+RUN make install
+RUN rm -rf /tmp/spdlog
+
+# build fmt
+WORKDIR /tmp
+RUN git clone --branch 11.1.3 https://github.com/fmtlib/fmt.git
+WORKDIR /tmp/fmt
+RUN mkdir build
+WORKDIR /tmp/fmt/build
+RUN cmake ..
+RUN make install
+RUN rm -rf /tmp/fmt
+
+# build flatbuffer
+WORKDIR /tmp
+RUN git clone --branch v25.2.10 https://github.com/google/flatbuffers.git
+WORKDIR /tmp/flatbuffers
+RUN cmake -G "Unix Makefiles" && \
+    make && \
+    make install
+
+# Set up environment variables for CUDA (optional if you plan to use CUDA)
+ENV PATH=/usr/local/flatbuffers/bin:$PATH
+RUN rm -rf /tmp/flatbuffers
+
+# Build boost
+ENV BOOST_VERSION="1.82.0"
+ENV BOOST_VERSION_UNDER="1_82_0"
+ENV BOOST_URL="https://archives.boost.io/release/${BOOST_VERSION}/source/boost_${BOOST_VERSION_UNDER}.tar.gz"
+ENV BOOST_BUILD_DIR=/tmp/boost_build
+ENV INSTALL_PREFIX=/usr/local
+
+# 创建构建目录
+RUN mkdir -p ${BOOST_BUILD_DIR}
+WORKDIR ${BOOST_BUILD_DIR}
+
+# 安装构建依赖
+# 下载和解压 Boost
+RUN curl -O ${BOOST_URL}
+RUN tar xzf boost_${BOOST_VERSION_UNDER}.tar.gz
+WORKDIR  boost_${BOOST_VERSION_UNDER}
+
+# 配置 Boost
+# 注意：这里使用 layout=system 以确保库文件安装在标准位置
+RUN ./bootstrap.sh --prefix=${INSTALL_PREFIX} --with-libraries=system,filesystem,thread,program_options,chrono,regex
+
+RUN ./b2 install \
+    variant=release \
+    link=static \
+    runtime-link=shared \
+    threading=multi \
+    --prefix=$INSTALL_PREFIX \
+    -j$(nproc) \
+    --layout=system
+# Somehow the above doesn't build boost_stacktrace_basic, so install it
+RUN dnf install -y boost boost-devel
+
+# Install pybind11 & python3 config
+ENV DEBIAN_FRONTEND=noninteractive
+
+# Install necessary build dependencies and dependencies for Python 3.12
+RUN dnf install -y python3.12 python3.12-pip
```

Comment:
manylinux pre-installed python. such as /opt/python/cp312-cp312/bin/python3.12. you can try to build wheel package by this script build_manylinux_wheels.sh